docs/how-to-docs/keycloak-sidecar.md

```diff
@@ -0,0 +1,109 @@
+
+## Keycloak sidecar using single role
+
+Below is a example of a keycloak sidecar which uses a single role for all URIs.In this example the configuration options are passed in through environment variables and command arguments.
+
+```yaml
+- name: keycloak-gatekeeper
+  image: quay.io/keycloak/keycloak-gatekeeper:latest
+  ports:
+    - containerPort: 10443
+      name: https
+      protocol: TCP
+  env:
+      # the secret of the client you are using. You can find this secret under the credentials tab of the client
+    - name: PROXY_CLIENT_SECRET
+      valueFrom:
+        secretKeyRef:
+          key: client_secret
+          name: gatekeeper
+  args:
+    # specifies what port the keycloak-gatekeeper is listening on
+    - --listen=:10443
+    # specififes which client is used
+    - --client-id=gatekeeper
+    # the url which is used to retrieve the OpenID configuration
+    - --discovery-url=https://sso-dev.notprod.homeoffice.gov.uk/auth/realms/hod-test
+    # the endpoint where requests are proxied to
+    - --upstream-url=https://127.0.0.1:10444
+    # The location of the certificate you wish the proxy to use for TLS support
+    - --tls-cert=/certs/tls.crt
+    # The location of a private key for TLS
+    - --tls-private-key=/certs/tls.key
+    # URls that you wish to protect.
+    - --resources=uri=/*
+  resources:
+    limits:
+      memory: 100Mi
+    requests:
+      memory: 50Mi
+  securityContext:
+    procMount: Default
+    runAsNonRoot: true
+    runAsUser: 1000
+  terminationMessagePath: /dev/termination-log
+  terminationMessagePolicy: File
+  volumeMounts:
+    - mountPath: /certs
+      name: certs
+    - mountPath: /etc/ssl/certs
+      name: bundle
+      readOnly: true
+```
+
+The above configuration uses the default roles for that realm. to use a specific role replace the `--resources` line with `--resources=uri=/*|roles=<name of role>`
+
+## Keycloak sidecar using multiple roles
+
+You can also use multiple roles for different URIs within your application. An example of this can be found below.
+
+```yaml
+- name: keycloak-gatekeeper
+  image: quay.io/keycloak/keycloak-gatekeeper:latest
+  ports:
+    - containerPort: 10443
+      name: https
+      protocol: TCP
+  env:
+    - name: PROXY_CLIENT_SECRET
+      valueFrom:
+        secretKeyRef:
+          key: client_secret
+          name: gatekeeper
+  args:
+    # specifies what port the keycloak-gatekeeper is listening on
+    - --listen=:10443
+    # specififes which client is used
+    - --client-id=gatekeeper
+    # the url which is used to retrieve the OpenID configuration
+    - --discovery-url=https://sso-dev.notprod.homeoffice.gov.uk/auth/realms/hod-test
+     # the endpoint where requests are proxied to
+    - --upstream-url=https://127.0.0.1:10444
+    - --tls-cert=/certs/tls.crt
+    - --tls-private-key=/certs/tls.key
+    # A user trying to access these URIs must have these roles to be successful
+    - --resources=uri=/users/*|roles=users
```

Comment:
I'd rather have `- --resources=uri=/*|roles=users` above to show that there is no need for distinct paths, i.e. it will work even if there is an overlap as keycloak will use the more specific definition

Reply:
made the change